src/Rector/v8/v7/MoveForeignTypesToOverrideChildTcaRector.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ssch\TYPO3Rector\Rector\v8\v7;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\ArrayItem;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Scalar\String_;
+use PhpParser\Node\Stmt\Return_;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Ssch\TYPO3Rector\Helper\TcaHelperTrait;
+use TYPO3\CMS\Core\Utility\ExtensionManagementUtility;
+
+/**
+ * @see https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/8.7/Deprecation-80000-InlineOverrideChildTca.html?highlight=foreign_types
+ */
+final class MoveForeignTypesToOverrideChildTcaRector extends AbstractRector
+{
+    use TcaHelperTrait;
+
+    /**
+     * @var string
+     */
+    private const FOREIGN_TYPES = 'foreign_types';
+
+    /**
+     * @var string
+     */
+    private const OVERRIDE_CHILD_TCA = 'overrideChildTca';
+
+    /**
+     * @codeCoverageIgnore
+     */
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('TCA InlineOverrideChildTca', [
+            new CodeSample(<<<'PHP'
+return [
+    'columns' => [
+        'aField' => [
+            'config' => [
+                'type' => 'inline',
+                'foreign_types' => [
+                    'aForeignType' => [
+                        'showitem' => 'aChildField',
+                    ],
+                ],
+            ],
+        ],
+    ],
+];
+PHP
+                , <<<'PHP'
+return [
+    'columns' => [
+        'aField' => [
+            'config' => [
+                'type' => 'inline',
+                'overrideChildTca' => [
+                    'types' => [
+                        'aForeignType' => [
+                            'showitem' => 'aChildField',
+                        ],
+                    ],
+                ],
+            ],
+        ],
+    ],
+];
+PHP
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Return_::class];
+    }
+
+    /**
+     * @param Return_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isTca($node)) {
+            return null;
+        }
+
+        $columns = $this->extractColumns($node);
+
+        if (! $columns instanceof ArrayItem) {
+            return null;
+        }
+
+        if (! $columns->value instanceof Array_) {
+            return null;
+        }
+
+        foreach ($this->extractColumnConfig($columns->value) as $columnConfig) {
+            //handle the special case of ExtensionManagementUtility::getFileFieldTCAConfig
+            if ($columnConfig instanceof StaticCall) {
```

Comment:
Would create a dedicated method here for better readibility.

Reply:
:+1: